### PR TITLE
FIR deserializer: load annotations for property and fields, along with use-site targets

### PR DESCRIPTION
**Pull request by jsjeon**
This applies a similar idea of https://github.com/JetBrains/kotlin/pull/3464 and https://github.com/JetBrains/kotlin/pull/3474 (which were for property accessors, functions, and constructors) to property and fields. But this time, use-site targets for loaded annotations are properly set.

### CODE REVIEW

**Comment by jsjeon** [comment]
@mglukhikh muted tests at https://github.com/JetBrains/kotlin/commit/b839a910500a6cd8df5b59218a75426327d35a25 are unmuted. Maybe ask a review to @dzharkov one more time?

**Comment by jsjeon** [comment]
Just a heads-up: it seems there is one regression on master.

File: compiler/fir/java/src/org/jetbrains/kotlin/fir/java/deserialization/JvmBinaryAnnotationDeserializer.kt [hunk 1]
```diff
@@ -118,7 +170,8 @@ class JvmBinaryAnnotationDeserializer(
                         if (signature.hasGetter()) MemberSignature.fromMethod(nameResolver, signature.getter) else null
                     CallableKind.PROPERTY_SETTER ->
                         if (signature.hasSetter()) MemberSignature.fromMethod(nameResolver, signature.setter) else null
-                    // TODO: PROPERTY
+                    CallableKind.PROPERTY ->
+                        getPropertySignature(proto, nameResolver, typeTable, true, true, requireHasFieldFlagForField)
```
Comment (juan-chen) [inline compiler/fir/java/src/org/jetbrains/kotlin/fir/java/deserialization/JvmBinaryAnnotationDeserializer.kt:174]: no need to pass the last two arguments when field is true?

Comment (jsjeon) [reply compiler/fir/java/src/org/jetbrains/kotlin/fir/java/deserialization/JvmBinaryAnnotationDeserializer.kt]: Thank you for spotting this! You're right; if `field` is true, `synthetic` is not necessary. Added https://github.com/JetBrains/kotlin/pull/3493/commits/d86d3b8d8f46dc67da10e0e4c105744d61e5b104 to change FE 1.0 part as well.

File: core/descriptors.jvm/src/org/jetbrains/kotlin/load/kotlin/AbstractBinaryClassAnnotationAndConstantLoader.kt [hunk 2]
```diff
@@ -397,7 +397,11 @@ abstract class AbstractBinaryClassAnnotationAndConstantLoader<A : Any, C : Any>(
                     AnnotatedCallableKind.PROPERTY_SETTER ->
                         if (signature.hasSetter()) MemberSignature.fromMethod(nameResolver, signature.setter) else null
                     AnnotatedCallableKind.PROPERTY ->
-                        getPropertySignature(proto, nameResolver, typeTable, true, true, requireHasFieldFlagForField)
+                        getPropertySignature(
```
[the remark this answers is not in the document]
Comment (jsjeon) [reply core/descriptors.jvm/src/org/jetbrains/kotlin/load/kotlin/AbstractBinaryClassAnnotationAndConstantLoader.kt:400]: As a context, I intentionally left @juan-chen 's comment above unfolded. As she mentioned, if `field` is true, `synthetic` is not used at all. Besides, it's better to use parameter name for boolean literals I believe.

Many parts of FIR deserializers are borrowed from the counter parts in FE 1.0, and I think it's good to update both if possible? Anyway, this one is the last commit that I intentionally put it as a separate one. I can squash it while dropping FE 1.0 part if you think we don't need this.

[the remark this answers is not in the document]
Comment (jsjeon) [reply core/descriptors.jvm/src/org/jetbrains/kotlin/load/kotlin/AbstractBinaryClassAnnotationAndConstantLoader.kt]: Okay, maybe it's better to minimize or isolate this PR's affected area. I partially reverted this part in `core`.

**Comment by jsjeon** [comment]
Rebased on top of recently added deserialization of value parameter.

**Comment by dzharkov** [comment]
Thank you for your contribution!
Merged manually

**Pull request closed without merging**